crates/ide/src/syntax_highlighting/highlight.rs

```diff
@@ -375,11 +375,11 @@ fn highlight_def(db: &RootDatabase, krate: Option<hir::Crate>, def: Definition)
                 if let Some(item) = func.as_assoc_item(db) {
                     h |= HlMod::Associated;
                     match func.self_param(db) {
-                        Some(sp) => {
-                            if let hir::Access::Exclusive = sp.access(db) {
-                                h |= HlMod::Mutable;
-                            }
-                        }
+                        Some(sp) => match sp.access(db) {
+                            hir::Access::Exclusive => h |= HlMod::Mutable,
```

Comment:
```suggestion
                            hir::Access::Exclusive => h = h | HlMod::Mutable | HlMod::Reference,
```
(can't use `h |= ...` due to a missing impl for that I believe)